### PR TITLE
updating docstring formatting for run_curve_fitting()

### PR DESCRIPTION
**Pull request by pavi-rajes**
Updating docstrings in run_curvefitting functions to include other reference functions (For example: get_modulationdepth() documentation should have a see also: run_curvefitting() clickable link)

### CODE REVIEW

File: aopy/analysis.py [hunk 1]
```diff
@@ -74,19 +74,21 @@ def factor_analysis_dimensionality_score(data_in, dimensions, nfold, maxiter=100
 
 
 # These functions are for curve fitting and getting modulation depth and preferred direction from firing rates
-def func(target, b1, b2, b3):
+def tuning_fit_func(target, b1, b2, b3):
```
Comment (leoscholl) [inline aopy/analysis.py:77]: you changed the name of func to tuning_fit_func, but you forgot to change it in run_curvefitting. a good example of why unit tests are important!

File: aopy/analysis.py [hunk 2]
```diff
@@ -74,19 +74,21 @@ def factor_analysis_dimensionality_score(data_in, dimensions, nfold, maxiter=100
...
     '''
-
+    Tuning function to be used in :func:`aopy.analysis.run_curvefitting`
     Args:
```
Comment (leoscholl) [inline aopy/analysis.py:80]: I think you need a newline above Args:

File: aopy/analysis.py [hunk 3]
```diff
@@ -74,19 +74,21 @@ def factor_analysis_dimensionality_score(data_in, dimensions, nfold, maxiter=100
...
 
     .. math::
-    
+
         b1 * cos(\\theta) + b2 * sin(\\theta) + b3
 
     Returns: result from above equation
 
+    See Also: :func:`aopy.analysis.run_curvefitting` , :func:`aopy.analysis.get_preferred_direction`, :func:`aopy.analysis.get_modulation_depth`
+
```
Comment (leoscholl) [inline aopy/analysis.py:91]: I think an example would be really helpful. That's also a benefit of having a unit test, so we can see how you are supposed to call the functions

File: aopy/analysis.py [hunk 4]
```diff
@@ -95,9 +97,14 @@ def func(target, b1, b2, b3):
 def get_modulation_depth(b1, b2):
     '''
     Calculates modulation depth from curve fitting parameters as follows:
-    
+    Args:
+        b1, b2 (float) : values from curve fitting
+    returns:
+        modulation direction from curve fitting parameters
+    See Also: :func:`aopy.analysis.get_preferred_direction`, :func:`aopy.analysis.run_curvefitting`, :func:`aopy.analyze.tuning_fit_func`
```
Comment (leoscholl) [inline aopy/analysis.py:104]: need newlines between each header. and Returns: with a capital R

File: aopy/analysis.py [hunk 5]
```diff
@@ -107,11 +114,16 @@ def get_modulation_depth(b1, b2):
...
     '''
     Calculates preferred direction from curve fitting parameters as follows:
-    
-    .. math:: 
-        
+    Args:
+        b1, b2 (float) : values from curve fitting
+    returns:
+        preferred direction from curve fitting parameters
+    See Also: See Also: :func:`aopy.analysis.run_curvefitting` , :func:`aopy.analysis.tuning_fit_func`, :func:`aopy.analysis.get_modulation_depth`
```
Comment (leoscholl) [inline aopy/analysis.py:121]: Same as above

File: aopy/analysis.py [hunk 6]
```diff
@@ -147,10 +159,11 @@ def run_curvefitting(means, make_plot=True, fig_title='Tuning Curve', n_subplot_
         n_cols (int) : No. of cols in subplot
 
     Returns:
-        tuple: Tuple containing:
-            | **params_day (Numpy array):** Curve fitting parameters
-            | **modulation depth (Numpy array):** Modulation depth of neuron
-            | **preferred direction (Numpy array):** preferred direction of neurons
+        params_day (Numpy array) : Curve fitting parameters
+        modulation depth (Numpy array) : Modulation depth of neuron
+        preferred direction (Numpy array) : preferred direction of neurons
```
Comment (leoscholl) [inline aopy/analysis.py:164]: you should keep the `| **` stuff, that helps with formatting on the website. 

also, can you please use more descriptive types than "numpy array"? what is the size of each array (n_days, n_neurons, nch, etc.)

File: aopy/analysis.py [hunk 7]
```diff
@@ -147,10 +159,11 @@ def run_curvefitting(means, make_plot=True, fig_title='Tuning Curve', n_subplot_
...
+
+    See Also: :func:`aopy.analysis.tuning_fit_func` , :func:`aopy.analysis.get_preferred_direction`, :func:`aopy.analysis.get_modulation_depth`
```
Comment (leoscholl) [inline aopy/analysis.py:166]: An example here would be appropriate. Maybe more useful than an example for any of the helper functions.

File: aopy/analysis.py [hunk 8]
```diff
@@ -147,10 +159,11 @@ def run_curvefitting(means, make_plot=True, fig_title='Tuning Curve', n_subplot_
         n_cols (int) : No. of cols in subplot
```
Comment (leoscholl) [inline aopy/analysis.py:159]: There's no description for this function. Can you tell us what it does? Including that it calls three other functions

**Review by leoscholl — COMMENTED** [review]
it's getting there. i strongly recommend you include unit tests in this pull request. it will count as two chores since there is a separate issue for including tests which you can also close... if you want to do it as a separate pull request that's ok too.

i recommend you read through the instructions on writing code/docstrings/tests again. all these errors are well documented there. https://analyze.readthedocs.io/en/latest/contributing.html

**Comment by pavi-rajes** [comment]
closing this pull request since it is old and most of the suggested changes have been addressed in other PR

**Pull request closed without merging**